### DISCUSSION TITLE
Class "Filament\Actions\Imports\Exceptions\RowImportFailedException" not found

### DISCUSSION THREAD

**Discussion by AHMED-GAMAL-AG** [filamentphp/filament#12950] [Help] [answered]
### Package

Other

### Package Version

3.2

### How can we help you?

I am trying to use the RowImportFailedException but I can't import the class I get class not found 

```
use App\Models\Truck;
use Filament\Actions\Imports\Exceptions\RowImportFailedException;

    public function resolveRecord(): ?Truck
    {
        $truck = Truck::where('plate_number', $this->data['plate_number'])->first();

        if ($this->data['truck_km'] < $truck->truck_km) {
            throw new RowImportFailedException(__('The truck end km must be greater than the current truck km.'));
        }

        return new Truck();
    }
```

![image](https://github.com/filamentphp/filament/assets/76778937/75f7d43d-1537-4e85-a95d-9eca32f8ccf9)

![image](https://github.com/filamentphp/filament/assets/76778937/13fd1b43-ff69-4b0d-b934-10238d51f9b4)

**Comment by KWangechi** [comment]
What if you use the full path? Like so: throw new \Filament\Actions\Imports\Exceptions\RowImportFailedException() ;

**Reply by AHMED-GAMAL-AG** [reply]
@KWangechi I get not found also, I checked the path in the vendor and found that the RowImportFailedException class doesn't exist in this path that's weird. Do you have any idea how to solve this? 

![image](https://github.com/filamentphp/filament/assets/76778937/1c26e455-f840-43ea-ad70-0cd6469efcf1)

**Reply by AHMED-GAMAL-AG** [reply]
do I need to open an issue or I am doing something wrong here? @danharrin

**Reply by danharrin** [reply]
What is your Filament version? `php artisan about`

**Reply by AHMED-GAMAL-AG** [reply]
The filament version is 3.2 

![image](https://github.com/filamentphp/filament/assets/76778937/74a51513-34af-474a-8c99-5394328d8c65)

**Reply by danharrin** [reply, marked as answer]
Composer.json should contain `^3.2` not `3.2` or `3.2.0`, because you're currently locked on a very old version without this feature.

**Reply by AHMED-GAMAL-AG** [reply]
Great that's fixed the problem for me thanks.
